src/app/static/src/app/store/adr/actions.ts

```diff
@@ -144,32 +144,49 @@ export const actions: ActionTree<ADRState, RootState> & ADRActions = {
 
-        for (let i = 0; i < uploadFilesPayload.length; i++) {
-            commit({type: ADRMutation.ADRUploadProgress, payload: i + 1});
-            const { resourceType, resourceFilename, resourceId } = uploadFilesPayload[i]
+        if (selectedDatasetId){
+            commit({type: ADRMutation.ADRUploadStarted, payload: uploadFilesPayload.length});
 
-            const requestParams: Dict<string> = {resourceFileName: resourceFilename}
-            if (resourceId){
-                requestParams["resourceId"] = resourceId
-            }
+            for (let i = 0; i < uploadFilesPayload.length; i++) {
+                commit({type: ADRMutation.ADRUploadProgress, payload: i + 1});
+                const { resourceType, resourceFilename, resourceId } = uploadFilesPayload[i]
 
-            let apiRequest = api<ADRMutation, ADRMutation>(context)
-                                        .withError(ADRMutation.SetADRUploadError);
-            if  (i === uploadFilesPayload.length - 1) {
-                apiRequest = apiRequest.withSuccess(ADRMutation.ADRUploadCompleted);
-            } else {
-                apiRequest = apiRequest.ignoreSuccess();
-            }
-            const response = await apiRequest.postAndReturn(`/adr/datasets/${selectedDatasetId}/resource/${resourceType}/${modelCalibrateId}`,
-                qs.stringify(requestParams))
-            if (!response) {
-                break
+                const requestParams: Dict<string> = {resourceFileName: resourceFilename}
+                if (resourceId){
+                    requestParams["resourceId"] = resourceId
+                }
+
+                let apiRequest = api<ADRMutation, ADRMutation>(context)
+                                            .withError(ADRMutation.SetADRUploadError);
+                if  (i === uploadFilesPayload.length - 1) {
+                    apiRequest = apiRequest.withSuccess(ADRMutation.ADRUploadCompleted);
+                } else {
+                    apiRequest = apiRequest.ignoreSuccess();
+                }
+                const response = await apiRequest.postAndReturn(`/adr/datasets/${selectedDatasetId}/resource/${resourceType}/${modelCalibrateId}`,
+                    qs.stringify(requestParams))
+                if (!response) {
+                    break
+                }
             }
+            let datasets = state.datasets;
+            await api(context)
+                .ignoreErrors()
+                .ignoreSuccess()
+                .get(`/adr/datasets/${selectedDatasetId}`)
+                .then((response) => {
+                    if (response) {
+                        datasets = [response.data];
+                    }
+                });
+    
+            const regenDataset = datasetFromMetadata(selectedDatasetId, datasets, state.schemas!);
+            commit(`baseline/${BaselineMutation.SetDataset}`, regenDataset, {root: true});
+            dispatch("getUploadFiles");
```

Comment:
I'm not completely of the right approach to this, but feels like the DownloadResults page should be the one dispatching this action, based on it watching changes to the baseline state. Otherwise we could end up with a tangle of actions calling other actions. Any thoughts on thiis @EmmaLRussell?

Reply:
I think by and large I'd rather have actions dispatching actions which should always always follow on from them, rather than relying on components to have that knowledge - the exception would be a case where for some reason the context of a particular component's call would require the follow-on action, but in other context it would not be required. We already use this pattern elsewhere (e.g. in baseline actions related to uploads).

Completely agree that we should avoid it becoming a tangle though and limit length of action 'chains' where possible.

Reply:
Ok - makes sense, thanks. The thought arose because in reality many of these actions are specific to a particular component, so when the actions call each other it appears to be coupling the components. But the alternative would mean more action at a distance - which isn't necessarily easier to manage.